apps/antalmanac/src/components/RightPane/SectionTable/GradesPopup.tsx

```diff
@@ -1,116 +1,137 @@
-import { Theme } from '@material-ui/core';
-import { withStyles } from '@material-ui/core/styles';
-import { ClassNameMap, Styles } from '@material-ui/core/styles/withStyles';
-import { Skeleton } from '@material-ui/lab';
-import { useState } from 'react';
+import { useState, useEffect, useMemo } from 'react';
 import { Bar, BarChart, CartesianGrid, ResponsiveContainer, XAxis, YAxis } from 'recharts';
+import { Box, Link, Typography, Skeleton } from '@mui/material';
+import { isDarkMode } from '$lib/helpers';
+import GradesHelper, { type Grades } from '$lib/grades';
 
-import { isDarkMode, queryGrades } from '$lib/helpers';
-
-const styles: Styles<Theme, object> = {
-    button: {
-        backgroundColor: '#385EB1',
-        color: '#fff',
-    },
-    gpaTitle: {
-        marginTop: '.5rem',
-        textAlign: 'center',
-        fontWeight: 500,
-        fontSize: '1.2rem',
-        marginRight: '4rem',
-        marginLeft: '4rem',
-    },
-    skeleton: {
-        padding: '4px',
-    },
-};
-
-interface GradesPopupProps {
+export interface GradeData {
+    grades: {
+        name: string;
+        all: number;
+    }[];
+    courseGrades: Grades;
+}
+
+async function getGradeData(
+    deptCode: string,
+    courseNumber: string,
+    instructor: string
+): Promise<GradeData | undefined> {
+    const courseGrades = await GradesHelper.queryGrades(deptCode, courseNumber, instructor, false).catch((e) => {
+        console.error(e);
+        return undefined;
+    });
+
+    if (!courseGrades) {
+        return undefined;
+    }
+
+    /**
+     * Format data for displayiing in chart.
```

Comment:
Typo -- probably by me lol

```suggestion
     * Format data for displaying in chart.
```